src/adaptors/amqp/container.c

```diff
@@ -69,15 +69,34 @@ struct qd_session_t {
     sys_atomic_t    ref_count;
     pn_session_t   *pn_session;
     qd_link_list_t  q3_blocked_links;  ///< Q3 blocked if !empty
+    uint32_t remote_max_frame;
+};
 
-    // For outgoing session flow control. Never buffer more than out_window_limit bytes of data on the session before
-    // returning control to the proactor. This prevents memory bloat and allows proactor to send buffered data in a
-    // timely manner. The low watermark is used to unblock the session - do not resume writing to the session until the
-    // amount of available capacity has grown to at least the low watermark.
+// Session window limits
+//
+// A session incoming window determines how many incoming frames the session will accept across all incoming links. This
+// places a limit on the number of incoming data bytes that have to be buffered on the session (connection max-frame *
+// max incoming window frames). The local session incoming window configuration is determined by the maxFrameSize and
+// maxSessionFrames configuration attributes of an AMQP listener/connector.
+//
+// The remote peers session window must be honored when writing output to a sending link. In addition we limit the
+// amount of outgoing data that can be buffered on a session before control is returned to Proton. This is necessary to
+// improve latency and allow capacity sharing among all links on the session.
+//
+const size_t   qd_session_max_outgoing_bytes = 1048576;   // max buffered bytes on a session
+const size_t   qd_session_low_outgoing_bytes = 524288;    // low watermark for max buffered bytes
 
-    size_t  out_window_limit;
-    size_t  out_window_low_watermark;
-};
+const size_t qd_session_max_in_win_user    = (size_t) 8388608;    // AMQP application in window max bytes 8MB
```

Comment:
> AMQP application

It is not clear to me what AMQP application means

Reply:
@ganeshmurthy "user connection"?  Basically any connection via a listener with the role "normal". These would include connections from the control plane to do management stuff. All non router-2-router connections.